tgui/packages/tgui/interfaces/MarionetteRemote.js

```diff
@@ -0,0 +1,163 @@
+                            onClick={() => act("set_data", { new_data: "NORTH" })}
+                            icon="arrow-up"
+                            selected={entered_data === "NORTH"}
+                          />
+                          <Button
+                            onClick={() => act("set_data", { new_data: "SOUTH" })}
+                            icon="arrow-down"
+                            selected={entered_data === "SOUTH"}
+                          /><Button
+                            onClick={() => act("set_data", { new_data: "WEST" })}
+                            icon="arrow-left"
+                            selected={entered_data === "WEST"}
+                          /><Button
+                            onClick={() => act("set_data", { new_data: "EAST" })}
+                            icon="arrow-right"
+                            selected={entered_data === "EAST"}
+                          />
+                        </>)}
+                  </LabeledList.Item>
+                  <LabeledList.Item key="command" label="Command">
+                    <Button
+                      onClick={() => act("set_command", { new_command: "say" })}
+                      content="Say"
+                      selected={selected_command === "say"}
+                    />
+                    <Button
+                      onClick={() => act("set_command", { new_command: "emote" })}
+                      content="Emote"
+                      selected={selected_command === "emote"}
+                    />
+                    <Button
+                      onClick={() => act("set_command", { new_command: "step" })}
+                      content="Step"
+                      selected={selected_command === "step"}
+                    />
+                    <Button
+                      onClick={() => act("set_command", { new_command: "drop" })}
+                      content="Drop"
+                      selected={selected_command === "drop"}
+                    />
+                    <Button
+                      onClick={() => act("set_command", { new_command: "use" })}
+                      content="Use"
+                      selected={selected_command === "use"}
+                    />
+                    <Button
+                      onClick={() => act("set_command", { new_command: "shock" })}
+                      content="Shock"
+                      selected={selected_command === "shock"}
+                    />
+                  </LabeledList.Item>
+                  <LabeledList.Item key="action_heat" label="Heat Per Action">
+                    {selected_command === "shock" || selected_command === "drop" ? "HIGH"
+                      : selected_command === "step" ? "LOW" : "MEDIUM"}
+                  </LabeledList.Item>
+                </LabeledList>
+              </Flex.Item>
+            </Flex>
+          </Section>
+          <Section title="Implants" buttons={(
+            <Button
```

Comment:
Unless I'm mistaken, haven't looked to deep in the dm code, using timely `tgui_process.update_uis(src)`, there'd be no need for a refresh button.